controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -242,53 +243,28 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	// Upgrade takes precedence over other operations
 	if len(requireUpgrade) > 0 {
 		logger.Info("Upgrading Control Plane")
-		if err := r.upgradeControlPlane(ctx, cluster, kcp, requireUpgrade); err != nil {
-			logger.Error(err, "Failed to upgrade the Control Plane")
-			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedUpgrade", "Failed to upgrade the control plane: %v", err)
-			return ctrl.Result{}, err
-		}
-		// TODO: need to requeue if there are additional operations to perform
-		return ctrl.Result{}, nil
+		return r.upgradeControlPlane(ctx, cluster, kcp, logger)
 	}
 
-	// If we've made it this far, we don't need to worry about Machines that are older than kcp.Spec.UpgradeAfter
-	currentMachines := ownedMachines.Filter(internal.MatchesConfigurationHash(currentConfigurationHash))
-	numMachines := len(currentMachines)
+	// If we've made it this far, we we can assume that all ownedMachines are up to date
+	numMachines := len(ownedMachines)
 	desiredReplicas := int(*kcp.Spec.Replicas)
 
 	switch {
 	// We are creating the first replica
 	case numMachines < desiredReplicas && numMachines == 0:
 		// Create new Machine w/ init
 		logger.Info("Initializing control plane", "Desired", desiredReplicas, "Existing", numMachines)
-		result, err := r.initializeControlPlane(ctx, cluster, kcp)
-		if err != nil {
-			logger.Error(err, "Failed to initialize control plane")
-			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedInitialization", "Failed to initialize cluster %s/%s control plane: %v", cluster.Namespace, cluster.Name, err)
-		}
-		// TODO: return the error if it is unexpected and should cause an immediate requeue
-		return result, nil
+		return r.initializeControlPlane(ctx, cluster, kcp, logger)
```

Comment:
Are we setting context that is important to this logger needing to passed down or could we instead opt for a global logger and omit the need for an additional parameter?

Reply:
Good point...  I think I realized this long after I started passing the logger around. I'll work on cleaning that up.

Reply:
Done.